src/averageAges.js

```diff
@@ -56,6 +73,18 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  return people
+    .filter(child => onlyWithSon ? child.sex === 'm' : true)
```

Comment:
putting too much inside `return` - better keep it more explanatory
```
let mothers = people.filter(child => getMother(people, child));

if (onlyWithSom) {
  mothers = mothers.filter(child => child.sex === 'm')
}

return getAverageAge(mothers)
```

getAverageAge(mothers) - it's another comment - create a helper function to calculate avg ages from the array passed